### PR TITLE
Increase the length limit for primary key strings to 256KB

### PR DESCRIPTION
**Pull request by benjaminwinger**
This also removes gaps from the string overflow files, at the cost of sometimes having to read multiple pages for a relatively short string, though if it's preferable I could make it so that only strings which are too large to fit in one page get split across multiple pages (or strings greater than a certain length).

I've added tests for both copying and setting primary key strings.

### CODE REVIEW

File: src/storage/storage_structure/in_mem_file.cpp [hunk 1]
```diff
@@ -35,32 +35,56 @@ ku_string_t InMemFile::appendString(const char* rawString) {
...
+                nextOffsetInPageToAppend = 0;
+                nextPageIdxToAppend++;
+            }
+            auto numBytesToWriteInPage = std::min(static_cast<uint64_t>(remainingLength),
+                BufferPoolConstants::PAGE_4KB_SIZE - nextOffsetInPageToAppend);
+            pages[nextPageIdxToAppend]->write(nextOffsetInPageToAppend,
+                (uint8_t*)rawString + (length - remainingLength), numBytesToWriteInPage);
+            // Only do this the first time
+            if (static_cast<uint64_t>(remainingLength) == length) {
+                TypeUtils::encodeOverflowPtr(
```
Comment (ray6080) [inline src/storage/storage_structure/in_mem_file.cpp:56]: You seem to be leaving gaps here, which is inconsistent from `DiskOverflowFile`. I don't think it's necessary to leave gaps anymore.

File: src/storage/storage_structure/disk_overflow_file.cpp [hunk 2]
```diff
@@ -17,12 +20,25 @@ std::string DiskOverflowFile::readString(TransactionType trxType, const ku_strin
     } else {
         PageByteCursor cursor;
```
[the remark this answers is not in the document]
Comment (benjaminwinger) [reply src/storage/storage_structure/disk_overflow_file.cpp:21]: Sounds good to me.

File: test/test_files/update_node/set_empty.test
[the remark this answers is not in the document]
Comment (benjaminwinger) [reply test/test_files/update_node/set_empty.test:1]: Sure. The error handing turned out to be somewhat poor in this case (it was throwing an exception in the InMemOverflowBuffer which might not be obvious to a user, so I added a custom exception at an earlier point).
I don't think the test actually gets to the point where it's relevant that it's a primary key though.

File: src/storage/storage_structure/in_mem_file.cpp (hunk 1 above)
Comment (benjaminwinger) [reply src/storage/storage_structure/in_mem_file.cpp]: I've fixed that bug, and it revealed that I was missing changes to the HashIndexBuilder's in-memory lookup.

File: test/test_files/update_node/set_empty.test
[the remark this answers is not in the document]
Comment (benjaminwinger) [reply test/test_files/update_node/set_empty.test:1]: The test is segfaulting on windows. I'm not sure why.

**Review by ray6080 — APPROVED** [review]

File: test/test_files/update_node/set_empty.test
[the remark this answers is not in the document]
Comment (ray6080) [reply test/test_files/update_node/set_empty.test:1]: hmm, I'm not sure about this "SEH" exception. Could it be memory related issues? "given that the set error gets thrown well before we reach this code": what do you mean "this code" here? was this thrown inside column writes or the index writes?

**Pull request merged**